test/functional/test-string.js

```diff
@@ -102,3 +107,13 @@ describe('expandTemplate', () => {
     expect(expandTemplate('${loop1}', testGetter), 10).to.equal('${loop2}');
   });
 });
+
+describe('camelCaseToDash', () => {
+  it('should convert camelCase strings to dash-case strings', () => {
+    expect(camelCaseToDash('foo')).to.equal('foo');
+    expect(camelCaseToDash('fooBar')).to.equal('foo-bar');
+    expect(camelCaseToDash('fooBarBaz')).to.equal('foo-bar-baz');
+    // Not really valid camel case
+    expect(camelCaseToDash('FooBarBaz')).to.equal('Foo-bar-baz');
```

Comment:
can we add an example that is mixed with numbers.
and also an example of having consecutive uppercase letters.